File: 0382-linked-list-random-node/0382-linked-list-random-node.py
import random
from typing import Optional

class Solution:
    def __init__(self, head: Optional[ListNode]):
        self.arr = []
        temp = head
        while temp != None:
            self.arr.append(temp.val)
            temp = temp.next

    def getRandom(self) -> int:
        return random.choice(self.arr)


# Your Solution object will be instantiated and called as such:
# obj = Solution(head)
# param_1 = obj.getRandom()

File: 0382-linked-list-random-node/0382-linked-list-random-node.cpp
class Solution {
private:
    ListNode* head;

public:
    Solution(ListNode* head) {
        this->head = head;
    }
    int getRandom() {
        int chosen_value = 0;
        ListNode* curr = head;
        int i = 1;
        while (curr != nullptr) {
            if (rand() % i == 0)chosen_value = curr->val;
            curr = curr->next;
            i++;
        }
        
        return chosen_value;
    }
};